flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/JobStatusObserver.java

```diff
@@ -95,6 +107,143 @@ public boolean observe(FlinkResourceContext<R> ctx) {
         return false;
     }
 
+    /**
+     * Observe the exceptions raised in the job manager and take appropriate action.
+     *
+     * @param ctx the context with which the operation is executed
+     */
+    protected void observeJobManagerExceptions(FlinkResourceContext<R> ctx) {
+        var resource = ctx.getResource();
+        var operatorConfig = ctx.getOperatorConfig();
+        var jobStatus = resource.getStatus().getJobStatus();
+
+        try {
+            var jobId = JobID.fromHexString(jobStatus.getJobId());
+            // TODO: Ideally the best way to restrict the number of events is to use the query param
+            // `maxExceptions`
+            //  but the JobExceptionsMessageParameters does not expose the parameters and nor does
+            // it have setters.
+            var history =
+                    ctx.getFlinkService()
+                            .getJobExceptions(
+                                    resource, jobId, ctx.getDeployConfig(resource.getSpec()));
```

Comment:
We should simply use `ctx.getObserveConfig()` here that's what we use for accessing the currently running cluster (to observe it)

Reply:
For my understanding 
What is the difference between the two configs?

Reply:
On a high level the observedConfig corresponds to the running job while the deployConfig corresponds to the config of the current spec. When the user makes a spec change these 2 can differ, so maybe they change some port settings etc, so its important to use the observeConfig when interacting with the job in the observer